### ISSUE TITLE
Enhancement Request: Merge arc-overhang with tree support

### ISSUE THREAD

**Issue by The-Monkey-King** [stmcculloch/arc-overhang#13]
Right now arc-overhang prints as it pertains to object(s) overhangs. However large overhangs require much slower print speed and stronger cooling. And even with these conditions met, the overhang has a very real possibility of filament deformation before subsequent layers can mitigate the deformation.

In another area of support, trees creates multiple branches to support object overhangs with less extraneous filament than a normal support construct. However, at the end of the day, it still prints more material than arc-overhang. 

But if tree and arc were merged, an equation could be developed where the slicer would "look ahead" to see whether the arc-overhang would be attached to the object or set on top of a tree branch as a concentric circle "platter." 

The former (arc connected to object) would be created as a result of: 
**[arc-overhang shape] < or = [good known boundary width, length of filament material strength*]**

* filament material strength is a measure of how far out a single layer of a particular filament can go unsupported without sag or lift

The latter (concentric circle placed on top of tree support branch) would be created as a result of two formulas:
1. Rejection of the above formula
2. Generate arc-overhang (concentric circle) + tree shape from a good, known base up to the center of the overhang. However, this concentric circle "platter" must not exceed the good know boundary width, length. If it does, then create more branches and platters to share the supported overhang equidistantly.  

Yes, this prints support material than current arc-overhang, but it prints less (I'm thinking) than normal tree supports with the benefit of printing tree supports only when the known good deformation-free arc-overhang to an object is exceeded. Meaning object-attached arc-overhangs will not curl or lift due to material weaknesses related to size of overhangs. 

**tl;dr** - Basically add/modify current a slicer's "Overhang threshold" setting to add another menu item, "Overhang support type:" with selectable items - "Default" (uses defined support material style, no arcs), "Arc-overhang" (ignores filament material strength limitations), and "Arc+Tree" (prints arcs on objects when size is not exceeded, and tree platters when an overhang is larger)
